soc/xtensa/esp32/CMakeLists.txt

```diff
@@ -3,4 +3,148 @@
+      -o ${CMAKE_BINARY_DIR}/zephyr/${CONFIG_KERNEL_BIN_NAME}.bin
+      ${CMAKE_BINARY_DIR}/zephyr/${CONFIG_KERNEL_BIN_NAME}.elf)
+  endif()
+
+  add_dependencies(app EspIdfBootloader EspPartitionTable)
+
+  board_finalize_runner_args(esp32 "--esp-flash-bootloader=${espidf_build_dir}/bootloader/bootloader.bin")
+
+  board_finalize_runner_args(esp32 "--esp-flash-partition_table=${espidf_build_dir}/partitions_singleapp.bin")
+
+  board_finalize_runner_args(esp32 "--esp-boot-address=0x1000")
+
+  board_finalize_runner_args(esp32 "--esp-partition-table-address=0x8000")
+
+  board_finalize_runner_args(esp32 "--esp-app-address=0x10000")
+endif()
+
+if(CONFIG_BOOTLOADER_MCUBOOT)
+  include(ExternalProject)
+  set(mcuboot_dir ${ZEPHYR_BASE}/../bootloader/mcuboot)
+  set(mcuboot_esp_dir ${mcuboot_dir}/boot/espressif)
+  set(mcuboot_build_dir ${CMAKE_BINARY_DIR}/mcuboot)
+
+  file(COPY ${CMAKE_CURRENT_LIST_DIR}/mcuboot.defconfig DESTINATION ${mcuboot_build_dir})
+
+  dt_nodelabel(dts_partition_path NODELABEL "boot_partition")
+  dt_reg_size(mcub_size PATH ${dts_partition_path})
+  file(APPEND ${mcuboot_build_dir}/mcuboot.defconfig
+    "CONFIG_ESP_BOOTLOADER_SIZE=${mcub_size}\n")
+
+  dt_nodelabel(dts_partition_path NODELABEL "slot0_partition")
+  dt_reg_addr(img_0_off PATH ${dts_partition_path})
+  dt_reg_size(img_size PATH ${dts_partition_path})
+  file(APPEND ${mcuboot_build_dir}/mcuboot.defconfig
+    "CONFIG_ESP_APPLICATION_PRIMARY_START_ADDRESS=${img_0_off}\n"
+    "CONFIG_ESP_APPLICATION_SIZE=${img_size}\n")
+
+  dt_nodelabel(dts_partition_path NODELABEL "slot1_partition")
+  dt_reg_addr(img_1_off PATH ${dts_partition_path})
+  file(APPEND ${mcuboot_build_dir}/mcuboot.defconfig
+    "CONFIG_ESP_APPLICATION_SECONDARY_START_ADDRESS=${img_1_off}\n"
+  )
+
+  dt_nodelabel(dts_partition_path NODELABEL "scratch_partition")
+  dt_reg_addr(img_scr_off PATH ${dts_partition_path})
+  dt_reg_size(img_scr_size PATH ${dts_partition_path})
+  file(APPEND ${mcuboot_build_dir}/mcuboot.defconfig
+    "CONFIG_ESP_SCRATCH_OFFSET=${img_scr_off}\n"
+    "CONFIG_ESP_SCRATCH_SIZE=${img_scr_size}\n"
+  )
+
+  if(CONFIG_MCUBOOT_GENERATE_CONFIRMED_IMAGE)
+    set(app_bin ${CONFIG_KERNEL_BIN_NAME}.signed.confirmed.bin)
+  else()
+    set(app_bin ${CONFIG_KERNEL_BIN_NAME}.signed.bin)
+  endif()
+
+  board_finalize_runner_args(esp32
+    "--esp-flash-bootloader=${mcuboot_build_dir}/mcuboot_esp32.bin")
+  board_finalize_runner_args(esp32 "--esp-boot-address=0x1000")
+  board_finalize_runner_args(esp32 "--esp-app-address=${img_0_off}")
+  board_finalize_runner_args(esp32
+    "--esp-flash-app=${CMAKE_BINARY_DIR}/zephyr/${app_bin}")
+
+  add_custom_target(mcub_conf DEPENDS ${mcuboot_build_dir}/mcuboot.defconfig)
+
+  ExternalProject_Add(
+    McuBoot
+    SOURCE_DIR ${mcuboot_esp_dir}
+    BINARY_DIR ${mcuboot_build_dir}
+    CONFIGURE_COMMAND ""
+    ${CMAKE_COMMAND} -G${CMAKE_GENERATOR}
+    -DCMAKE_TOOLCHAIN_FILE=${CMAKE_CURRENT_LIST_DIR}/mcuboot-esp32.cmake
```

Comment:
adding a toolchain file, which doesn't really look like toolchain file because it doesn't configure anything regarding CMAKE_C_COMPILER or similar.
https://cmake.org/cmake/help/latest/manual/cmake-toolchains.7.html

It set some compile flags as cache variables, so why isn't that done as pre-load ?
`cmake -C <file>`
https://cmake.org/cmake/help/latest/manual/cmake.1.html#options

Note, even in that case, why does the `mcuboot-esp32.cmake` file belong in the Zephyr CMake build system.

It's not required in any way to build Zephyr itself.
This should be part of the TSC meeting.

Reply:
Not used, closing this thread.